### PR TITLE
feat: add support for non-deterministic models in GeneralSemanticRobustness and add BERTScore Dissimilarity

### PR DESCRIPTION
**Pull request by bilalaws**
*Description of changes:*

This PR implements the following changes to `GeneralSemanticRobustness` eval:

1. To complement the `Word Error Rate` metric that measures syntactic differences, we add the `BERTScore Dissimilarity` metric that measures semantic differences. We use `BERTScore Dissimilarity = 1 - BERTScore` (a _dissimilarity_ metric) instead of `BERTScore` (a _similarity_ metric). We use dissimilarity to be consistent with `Word Error Rate` and the rest of `SemanticRobustness` evals that measure dissimilarities.
2. We normalize the `BERTScore Dissimilarity` and `Word Error Rate` when the model is non-deterministic.

### CODE REVIEW

**Review by polaschwoebel — APPROVED** [review]
I approve. We had multiple rounds of offline discussion, no additional comments.

File: src/fmeval/constants.py [hunk 1]
```diff
@@ -121,3 +121,30 @@ class DatasetColumns(Enum):
 JUMPSTART_BUCKET_BASE_URL_FORMAT = "https://jumpstart-cache-prod-{}.s3.{}.amazonaws.com"
 JUMPSTART_BUCKET_BASE_URL_FORMAT_ENV_VAR = "JUMPSTART_BUCKET_BASE_URL_FORMAT"
 GENERATED_TEXT_JMESPATH_EXPRESSION = "*.output_keys.generated_text"
+
+
+class BertscoreModels(Enum):
```
[the remark this answers is not in the document]
Comment (bilalaws) [reply src/fmeval/constants.py:126]: Done.

File: test/unit/eval_algorithms/test_summarization_accuracy.py [hunk 2]
```diff
@@ -792,11 +795,11 @@ def test_get_rouge_score(self, test_case, load_meteor_helpers, config):
             ),
         ],
     )
-    @patch("fmeval.eval_algorithms.summarization_accuracy.ray.get")
+    @patch("fmeval.eval_algorithms.util.ray.get")
     def test_get_bert_score(self, mock_ray_get, test_case, config):
```
[the remark this answers is not in the document]
Comment (bilalaws) [reply test/unit/eval_algorithms/test_summarization_accuracy.py:757]: Done

**Review by polaschwoebel — APPROVED** [review]
No scientific changes since last review.

**Review by oyangz — CHANGES_REQUESTED** [review]
Could you add a short metric description in the constants [here](https://github.com/aws/fmeval/blob/main/src/fmeval/reporting/constants.py#L168) for reporting?

**Review by malhotra18 — APPROVED** [review]

**Pull request merged**